README.md

```diff
@@ -1,48 +1,97 @@
+
+## Package Summary
+The `test-creation` package aims to facilitate the creation of robust and reproducible software tests for applied machine learning code. It includes functionalities to ensure the presence, quality, and proper ingestion of data at the beginning of the analysis, the correct fitting and evaluation of models, and the integrity of artifacts generated during the analysis.
 
 ## Installation
 
-1. Create a conda envionment using `environment.yml` in the repo:
+### Install from PyPI
+
+Run this command to install the package
+
+```bash
+$ pip install test-creation
+```
+
+### Install from GitHub
+
+Before proceeding with this installation, ensure you have Miniconda/Anaconda installed on your system. These tools provide support for creating and managing Conda environments.
+
+#### Step 1: Clone the Repository
+
+Start by cloning the repository to your local machine. Open your terminal and run the following command:
 
 ```bash
-conda env create -f environment.yml
+$ git clone git@github.com:UBC-MDS/test-creation.git
 ```
 
-2. Activate the newly created conda environment (default name `test-creation`):
+#### Step 2: Create and Activate the Conda Environment
+
+Create a new Conda environment using the environment.yaml file provided in this repository. This file contains all the necessary dependencies, including both Python and Poetry versions.
+
+To create the environment, open your terminal and navigate to the directory where the environment.yaml file is located. Then, run the following command:
+
 ```bash
-conda activate test-creation
+$ conda env create -f environment.yaml
+$ conda activate test-creation
 ```
 
-3. add `.env` with API key attached:
+#### Step 3: Install the Package Using Poetry
+
+With the Conda environment activated, you can now use Poetry to install the package. Run the following command to install the package using Poetry:
+
 ```bash
-echo "OPENAI_API_KEY=..." > .env
+$ poetry install
 ```
 
-4. Enjoy!
+This command reads the pyproject.toml file in your project (if present) and installs the dependencies listed there.
 
-## Usage
+#### Running the tests
+Navigate to the project root directory and use the following command in terminal to test the functions defined in the projects. 
 
-### code_analyzer
+``` bash
+$ pytest tests/*
+```
 
-This is used to analyze and extract information from a downloaded repo.
+#### Troubleshooting
+Environment Creation Issues: If you encounter problems while creating the Conda environment, ensure that the environment.yaml file is in the correct directory and that you have the correct version of Conda installed.
 
-There is a file containing the example calls. Run:
+## Usage
 
 ```console
-$ python ./src/code_analyzer/example.py <path-to-your-repo>
+$ python ./src/test_creation/analyze.py <path-to-your-checklist> <path-to-your-repo>
```

Comment:
Have to separate and update this example. 1 for test evaluator, 1 for test spec generator